src/Avalonia.Base/Media/PolyBezierSegment.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using Avalonia.Utilities;
+
+namespace Avalonia.Media;
+
+/// <summary>
+/// PolyBezierSegment
+/// </summary>
+public sealed class PolyBezierSegment : PathSegment
+{
+    /// <summary>
+    /// Points DirectProperty definition
+    /// </summary>
+    public static readonly DirectProperty<PolyBezierSegment, Points?> PointsProperty =
+        AvaloniaProperty.RegisterDirect<PolyBezierSegment, Points?>(nameof(Points),
+            o => o.Points,
+            (o, v) => o.Points = v);
+
+    private Points? _points = [];
+
+    public PolyBezierSegment()
+    {
+
+    }
+
+    public PolyBezierSegment(IEnumerable<Point> points, bool isStroked)
+    {
+        if (points is null)
+        {
+            throw new ArgumentNullException(nameof(points));
+        }
+
+        Points = new Points(points);
+        IsStroked = isStroked;
+    }
+
+    /// <summary>
+    /// Gets or sets the Point collection that defines this <see cref="PolyBezierSegment"/> object.
+    /// </summary>
+    /// <value>
+    /// The points.
+    /// </value>
+    [Metadata.Content]
+    public Points? Points
+    {
+        get => _points;
+        set => SetAndRaise(PointsProperty, ref _points, value);
+    }
+
+    internal override void ApplyTo(StreamGeometryContext ctx)
+    {
+        var isStroken = this.IsStroked;
+        if (_points is { Count: > 0 } points)
+        {
+            var i = 0;
+            for (; i < points.Count; i += 3)
+            {
+                ctx.CubicBezierTo(points[i],
+                    points[i + 1],
+                    points[i + 2],
+                    isStroken);
+            }
+            var delta = i - points.Count;
```

Comment:
I saw that. I thought it would be helpful for developers to have feedback if they were using incorrect values ​​so they could more easily look for bugs.

Do I remove the warning?